src/table/Root.jsx

```diff
@@ -12,7 +13,9 @@ export function render(rootElement, props) {
       <StyledEngineProvider injectFirst>
         <StylesProvider generateClassName={muiParameters.generateClassName}>
           <ThemeProvider theme={muiParameters.theme}>
-            <TableWrapper {...props} />
+            <RootContext.Provider value={props}>
```

Comment:
there is two situations for providing the values to the component tree. 
1. one is providing it from here
2. the other one is providing it from Table wrapper, since we might need some stuff from inside of table wrapper to be available for us in our context


so i think it could be really better if we could have an isolated layer between `<ThemeProvider />` and `<TableWrapper/>` component for our context and all logics should goes inside of that. 

in future, if we want to switch for custom hooks for each of the works that we do for table (selection, sorting, pagination, a11y and ... ) that layer would helps us a lot, and it could be the single source for our logic. 

just consider the context layer as a hub, that gathers all of the required data variables and functions ( as well as utils hooks ) inside of the context itself and them memoizes  that and exposes into the component tree.